### PR TITLE
565987 API revision | implement license importing flow

### PR DESCRIPTION
**Pull request by eparovyshnaya**
- rename LicensedAplicationFromContext to `product`notation
- reimplement it's version part on current version support
- and test it

### CODE REVIEW

File: bundles/org.eclipse.passage.lic.e4.ui/src/org/eclipse/passage/lic/internal/e4/ui/addons/E4LicensingAddon.java [hunk 1]
```diff
@@ -30,6 +34,7 @@
 
 	private final IApplicationContext application;
 	private final IEclipseContext context;
+	private java.util.Optional<ExaminationCertificate> certificate = java.util.Optional.empty();
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.e4.ui/src/org/eclipse/passage/lic/internal/e4/ui/addons/E4LicensingAddon.java:37]: rock!

File: tests/org.eclipse.passage.lic.base.tests/src/org/eclipse/passage/lic/internal/base/tests/access/NoSevereRestrictionsTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package org.eclipse.passage.lic.internal.base.tests.access;
```
Comment (ruspl-afed) [inline tests/org.eclipse.passage.lic.base.tests/src/org/eclipse/passage/lic/internal/base/tests/access/NoSevereRestrictionsTest.java:1]: header

File: tests/org.eclipse.passage.lic.equinox.tests/src/org/eclipse/passage/lic/internal/equinox/tests/LicensedProductFromContextTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
+package org.eclipse.passage.lic.internal.equinox.tests;
```
Comment (ruspl-afed) [inline tests/org.eclipse.passage.lic.equinox.tests/src/org/eclipse/passage/lic/internal/equinox/tests/LicensedProductFromContextTest.java:1]: header

**Review by ruspl-afed — CHANGES_REQUESTED** [review]

File: bundles/org.eclipse.passage.lic.e4.ui/src/org/eclipse/passage/lic/internal/e4/ui/addons/E4LicensingAddon.java (hunk 1 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.lic.e4.ui/src/org/eclipse/passage/lic/internal/e4/ui/addons/E4LicensingAddon.java:37]: hate `null`s

File: tests/org.eclipse.passage.lic.base.tests/src/org/eclipse/passage/lic/internal/base/tests/access/NoSevereRestrictionsTest.java (hunk 2 above)
Comment (eparovyshnaya) [reply tests/org.eclipse.passage.lic.base.tests/src/org/eclipse/passage/lic/internal/base/tests/access/NoSevereRestrictionsTest.java:13]: added

File: tests/org.eclipse.passage.lic.equinox.tests/src/org/eclipse/passage/lic/internal/equinox/tests/LicensedProductFromContextTest.java (hunk 3 above)
Comment (eparovyshnaya) [reply tests/org.eclipse.passage.lic.equinox.tests/src/org/eclipse/passage/lic/internal/equinox/tests/LicensedProductFromContextTest.java:13]: done

**Review by ruspl-afed — APPROVED** [review]

**Pull request merged**